python/nano/tutorial/training/pytorch/pytorch_train_multi_instance.py

```diff
@@ -0,0 +1,117 @@
+                                          transforms.ToTensor(),
+                                          transforms.Normalize([0.485, 0.456, 0.406],
+                                                               [0.229, 0.224, 0.225])])
+    val_transform = transforms.Compose([transforms.Resize(256),
+                                        transforms.CenterCrop(224),
+                                        transforms.ToTensor(),
+                                        transforms.Normalize([0.485, 0.456, 0.406],
+                                                             [0.229, 0.224, 0.225])])
+
+    # Apply data augmentation to the tarin_dataset
+    train_dataset = OxfordIIITPet(root="/tmp/data", transform=train_transform, download=True)
+    val_dataset = OxfordIIITPet(root="/tmp/data", transform=val_transform)
+
+    # obtain training indices that will be used for validation
+    indices = torch.randperm(len(train_dataset))
+    val_size = len(train_dataset) // 4
+    train_dataset = torch.utils.data.Subset(train_dataset, indices[:-val_size])
+    val_dataset = torch.utils.data.Subset(val_dataset, indices[-val_size:])
+
+    # prepare data loaders
+    train_dataloader = DataLoader(train_dataset, batch_size=32)
+    val_dataloader = DataLoader(val_dataset, batch_size=32)
+
+    return train_dataloader, val_dataloader
+
+
+# subclass TorchNano and override its train() method
+class MyNano(TorchNano):
+    # move the body of your existing train function into TorchNano train method
+    def train(self):
+        seed_everything(42)
+        model = MyPytorchModule()
+        optimizer = torch.optim.SGD(model.parameters(), lr=0.01, momentum=0.9, weight_decay=5e-4)
+        loss_fuc = torch.nn.CrossEntropyLoss()
+        train_loader, val_loader = create_dataloaders()
+
+        # call `setup` to prepare for model, optimizer(s) and dataloader(s) for accelerated training
+        model, optimizer, (train_loader, val_loader) = self.setup(model, optimizer,
+                                                                  train_loader, val_loader)
+        num_epochs = 5
+
+        # EPOCH LOOP
+        for epoch in range(num_epochs):
+
+            # TRAINING LOOP
+            model.train()
+            train_loss, num = 0, 0
+            for data, target in train_loader:
+                optimizer.zero_grad()
+                output = model(data)
+                loss = loss_fuc(output, target)
+                # replace the loss.backward() with self.backward(loss)
+                self.backward(loss)
+                optimizer.step()
+
+                train_loss += loss.sum()
+                num += 1
+            print(f'Train Epoch: {epoch}, loss: {train_loss/num}')
+
+
+if __name__ == '__main__':
+    # Multi-instance Training
+    #
+    # It is often beneficial to use multiple instances
+    # for training if a server contains multiple sockets or
+    # many cores, so that the workload can make full use of
+    # all CPU cores.
```

Comment:
also need to explain learning rate scaling & warmup? <https://github.com/intel-analytics/BigDL/pull/5463#discussion_r963776002>

Reply:
Done